packages/block-editor/src/hooks/content-lock-ui.js

```diff
@@ -107,39 +107,51 @@ export const withBlockControls = createHigherOrderComponent(
 		return (
 			<>
 				{ isEditingAsBlocks && ! isContentLocked && (
-					<StopEditingAsBlocksOnOutsideSelect
-						clientId={ props.clientId }
-						stopEditingAsBlock={ stopEditingAsBlock }
-					/>
+					<>
+						<StopEditingAsBlocksOnOutsideSelect
+							clientId={ props.clientId }
+							stopEditingAsBlock={ stopEditingAsBlock }
+						/>
+						<BlockControls group="other">
+							<ToolbarButton
+								onClick={ () => {
+									stopEditingAsBlock();
+								} }
+							>
+								{ __( 'Done' ) }
+							</ToolbarButton>
+						</BlockControls>
+					</>
+				) }
+				{ ! isEditingAsBlocks && isContentLocked && (
```

Comment:
We'll need to include `isSelected` in the check; otherwise, the "Modify" menu item will be rendered multiple times. You can reproduce this by adding more than one content-only locked pattern to the post.

I used a similar method in the Template Parts block - https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/template-part/edit/index.js#L96-L99.

## Screenshot

![CleanShot 2022-10-31 at 10 36 59](https://user-images.githubusercontent.com/240569/198947286-f208b8fc-23e9-4add-a81d-b74d1952866f.png)

Reply:
Nice catch it is fixed 👍